sickbeard/subtitles.py

```diff
@@ -731,17 +731,22 @@ def subtitles_download_in_pp():  # pylint: disable=too-many-locals, too-many-bra
                 video_path = os.path.join(root, filename)
 
                 if processTV.subtitles_enabled(video_path) is False:
-                    logger.debug(u'Subtitle disabled for show: %s', filename)
+                    logger.debug(u'Subtitle disabled for show: %s. Running post-process to PP it', filename)
+                    run_post_process = True
                     continue
 
                 release_name = os.path.splitext(filename)[0]
                 found_subtitles = download_best_subs(video_path, root, release_name, languages, subtitles=False,
-                                                     embedded_subtitles=bool(not sickbeard.EMBEDDED_SUBTITLES_ALL),
+                                                     embedded_subtitles=False,
                                                      provider_pool=pool)
                 downloaded_languages = {s.language.opensubtitles for s in found_subtitles}
 
-                # Don't run post processor unless at least one file has all of the needed subtitles
-                if not run_post_process and not needs_subtitles(downloaded_languages):
+                # Don't run post processor unless at least one file has all of the needed subtitles OR
+                # if user don't want to ignore embedded subtitles and wants to consider 'unknown' as wanted sub,
+                # and .mkv has one.
+                if not run_post_process and ((not needs_subtitles(downloaded_languages)) or
+                    (not sickbeard.EMBEDDED_SUBTITLES_ALL and sickbeard.EMBEDDED_SUBTITLES_UNKNOWN_LANG
```

Comment:
This condition is exaclty the same that you have here: https://github.com/pymedusa/SickRage/pull/743/files#diff-f37f325022d82d8eb473757a1bfa43f3R573

Better to have it in a method and just call the method. processTV.matches_unkown_subtitles or something similar